File: algo/longest-common-prefix/longestCommonPrefix.py
class Solution(object):
    def longestCommonPrefix(self, strs):
        """
        :type strs: List[str]
        :rtype: str
        """
        if not strs:
            return ""
        
        last, lengths = 0, {}
        for x in xrange(1, len(strs)):
            lengths[x] = len(strs[x])
            
        for x in xrange(len(strs[0])):
            saved = strs[0][x]
            for y in xrange(1, len(strs)):
                if x >= lengths[y] or saved != strs[y][x]:
                    return strs[0][:last]
            last += 1
        return strs[0]

File: algo/longest-common-prefix/longestCommonPrefix.go
func longestCommonPrefix(strs []string) string {
    if len(strs) == 0 {
        return ""
    }
    
    end, length := 0, map[int]int{}
    
    for x := 1; x < len(strs); x++ {
        length[x] = len(strs[x])
    }
    
    for y := 0; y < len(strs[0]); y++ {
        saved := strs[0][y]
        for x := 1; x < len(strs); x++ {
            if length[x] <= y || strs[x][y] != saved {
                return strs[0][:end]
            }
        }
        end++
    }
    return strs[0]
}